conveyor/generator/generator/templates.go

```diff
@@ -0,0 +1,146 @@
+/*
+ *    Copyright 2019 Insolar Technologies
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package generator
+
+import (
+	"io"
+	"text/template"
+)
+
+var (
+	funcMap = template.FuncMap{
+		"inc": func(i int) int {
+			return i + 1
+		},
+		"notNil": func(x interface{}) bool {
+			return x == nil
+		},
+	}
+	stateMachineTpl = template.Must(template.New("stateMachineTpl").Funcs(funcMap).Parse(`
+package {{.Package}}
+
+import (
+	"github.com/insolar/insolar/conveyor/generator/common"
+	"github.com/insolar/insolar/conveyor/interfaces/slot"
+	"errors"
+)
+
+{{range $i, $machine := .StateMachines}}type SMFID{{$machine.Name}} struct {
+}
+func (*SMFID{{$machine.Name}}) TID() common.ElType {
+    return {{$i}}
+}
+{{range $i, $state := .States}}
+func (*SMFID{{$machine.Name}}) {{$state.Name}}() common.ElState {
+    return {{inc $i}}
+}{{end}}
+
+type SMRH{{$machine.Name}} struct {
+	cleanHandlers {{$machine.Name}}
+}
+
+func SMRH{{$machine.Name}}Export() common.StateMachine {
+    m := SMRH{{$machine.Name}}{
+        cleanHandlers: &{{$machine.Name}}Implementation{},
+    }
+    var x []common.State
+    x = append(x, common.State{
+	        Transit: m.Init,
+	        Error: m.{{$machine.InitError.Name}},
+        }, {{range $i, $state := .States}}
+        common.State{
+	        Transit: m.{{$state.Transit.Name}},
+	        Migrate: m.{{$state.Migrate.Name}},
+	        Error: m.{{$state.Error.Name}},
+        },
+    {{end}})
+    return common.StateMachine{
+        Id: int(m.cleanHandlers.({{$machine.Name}}).TID()),
+        States: x,
+    }
+}
+
+func (s *SMRH{{$machine.Name}}) Init(element slot.SlotElementHelper) (interface{}, uint32, error) {
+    aInput, ok := element.GetInputEvent().({{$machine.Init.EventType}})
+    if !ok {
+        return nil, 0, errors.New("wrong input event type")
+    }
+    payload, state, err := s.cleanHandlers.Init(aInput)
+    if err != nil {
+        return payload, state.ToInt(), err
```

Comment:
do you really need return payload and state.ToInt() in case of error?

Reply:
Of course